### PR TITLE
FunctionUtils polishing

### PR DESCRIPTION
**Pull request by olegz**
- attempted to make code more functional (eat our own dog food) and concise

### CODE REVIEW

File: spring-cloud-function-core/src/main/java/org/springframework/cloud/function/support/FunctionUtils.java [hunk 1]
```diff
@@ -110,59 +112,15 @@ public static boolean isFluxFunction(Function<?, ?> function) {
...
-			if (args != null) {
-				String[] typeNames = new String[args.length];
-				for (int i = 0; i < args.length; i++) {
-					typeNames[i] = args[i].getTypeName();
-				}
-				return typeNames;
-			}
-		}
-		return new String[0];
+	private static boolean isFlux(int lenth, String... types){
```
Comment (dsyer) [inline spring-cloud-function-core/src/main/java/org/springframework/cloud/function/support/FunctionUtils.java]: nit: spelling error "lenth"

**Comment by olegz** [comment]
@dsyer fixed, rebased and squashed

File: spring-cloud-function-core/src/main/java/org/springframework/cloud/function/support/ConfiguredFunctionUtils.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import reactor.core.publisher.Flux;
+
+/**
+ * <p>
+ * Miscellaneous utility operations to interrogate functional
+ * components (beans) configured in BeanFactory.
+ * </p>
+ * <p>
+ * It is important to understand that it is not a general purpose utility
+ * to interrogate “any” functional component. Certain operations may/will
```
Comment (dsyer) [inline spring-cloud-function-core/src/main/java/org/springframework/cloud/function/support/ConfiguredFunctionUtils.java]: Please use regular double quotes (")

Comment (olegz) [inline spring-cloud-function-core/src/main/java/org/springframework/cloud/function/support/ConfiguredFunctionUtils.java]: ouch! note to self; don't use Mail client to write javadocs for spell check

**Comment by olegz** [comment]
@dsyer all done

File: spring-cloud-function-context/src/test/java/org/springframework/cloud/function/context/ContextFunctionCatalogAutoConfigurationTests.java [hunk 3]
```diff
@@ -442,7 +442,10 @@ public static void set(Object value) {
 	protected static class MessageConfiguration {
 		@Bean
 		public Function<Message<String>, Message<String>> function() {
-			return m -> MessageBuilder.withPayload(m.getPayload().toUpperCase()).build();
+			return m -> {
+				System.out.println("Message: " + m);
```
Comment (markfisher) [inline spring-cloud-function-context/src/test/java/org/springframework/cloud/function/context/ContextFunctionCatalogAutoConfigurationTests.java]: this was left on accident?

**Pull request closed without merging**